### ISSUE TITLE
Lathe with STM32F411

### ISSUE THREAD

**Issue by tuxcnc** [grblHAL/STM32F4xx#180]
Hi.
I made some tests with BlackPill (411, firmware from Web Builder)  connected to two generators, one 3200 Hz as pulses, and second 1 Hz short pulses as index.
I surprised the G33 (and G76 too) works after M5 (spindle off), not after M3 (spindle on).
The second problem I found is why, and for what, the G33 checks first is spindle turns, and gives error if not?
The G33 has timeout for case of no index pulses, and in my opinion it is enough.
I think, the check is spindle turns prevents from G33.1 implementation, because there is direction change and G33 procedure must starts when spindle speed is zero...
Regards.

**Comment by terjeio** [comment]
> I think the signal pins are buried (on purpose) somewhere in the Grbl file maze.

Not really, the issue is that only a few pins are supported making it tricky to assign them. E.g. [this](https://github.com/grblHAL/STM32F4xx/blob/1cfbda75bd71d59eb73aa3c700baf5d526722e1a/boards/st_morpho_map.h#L249-L256) and [this](https://github.com/grblHAL/STM32F4xx/blob/1cfbda75bd71d59eb73aa3c700baf5d526722e1a/boards/blackpill_alt2_map.h#L148-L155) map file has definitions for them.

The `SPINDLE_PULSE_PIN` is the most limited, if GPIOD 2 can be used then that is preferable, if not a pin that can be routed as a counter input is required - and this is not always easy to achieve. The `SPINDLE_INDEX_PIN` can be any interrupt capable pin.
